src/Oracle/OracleService.cs

```diff
@@ -0,0 +1,485 @@
+using Akka.Actor;
+using Akka.Util.Internal;
+using Microsoft.Extensions.Configuration;
+using Neo.ConsoleService;
+using Neo.Cryptography;
+using Neo.Cryptography.ECC;
+using Neo.IO;
+using Neo.IO.Json;
+using Neo.Ledger;
+using Neo.Network.P2P;
+using Neo.Network.P2P.Payloads;
+using Neo.Persistence;
+using Neo.SmartContract;
+using Neo.SmartContract.Native;
+using Neo.VM;
+using Neo.Wallets;
+using Neo.Wallets.NEP6;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Timers;
+using static System.IO.Path;
+using NJArray = Newtonsoft.Json.Linq.JArray;
+using NJObject = Newtonsoft.Json.Linq.JObject;
+using NUtility = Neo.Utility;
+
+namespace Neo.Plugins
+{
+    public class OracleService : Plugin, IPersistencePlugin
+    {
+        private NEP6Wallet Wallet;
+        private string[] Nodes;
+        private TimeSpan MaxTaskTimeout;
+        private ConcurrentDictionary<ulong, OracleTask> PendingQueue;
+        private CancellationTokenSource CancelSource;
+        private int Counter;
+        private ConcurrentDictionary<ulong, DateTime> FinishedCache;
+        private ConsoleServiceBase ConsoleBase;
+
+        private static readonly object _lock = new object();
+
+        private const int RefreshInterval = 1000 * 60 * 3;
+
+        private static readonly OracleHttpProtocol Https = new OracleHttpProtocol();
+
+        public override string Description => "Built-in oracle plugin";
+
+        public OracleService()
+        {
+            PendingQueue = new ConcurrentDictionary<ulong, OracleTask>();
+            FinishedCache = new ConcurrentDictionary<ulong, DateTime>();
+
+            RpcServerPlugin.RegisterMethods(this);
+
+            System.Timers.Timer timer = new System.Timers.Timer();
```

Comment:
It's better to move this variable outside the method, otherwise you can't stop the service.

Reply:
Done